### DISCUSSION TITLE
Do we need call stream_->stop if stream_->requestStart failed

### DISCUSSION THREAD

**Discussion by xzhan96** [google/oboe#1450] [Q&A] [answered]
Hi Experts

Could you help give some help on below question?

Even stream_->requestStart failed, a valid stream may already be opened, do we need call  stream_->stop  and  stream_->close explicitly, or just leave it and let oboe release it automatically?

**Comment by philburk** [comment, marked as answer]
You don't need to stop it because it failed to start.
But since the open succeeded you should close it.

You should always close() streams as soon as you are done using them so that other apps can have access to the audio resources.
